src/util/Spotify.js

```diff
@@ -48,53 +48,66 @@ export const Spotify = {
         Tracklist.js, cannot read property 'map' of undefined, and pointing on the
         traks of the TrackList.. I couldn't understand why.*/
     async search (term) {
-      /* let userAccessToken = this.getUserAccessToken();   I commented this out becaues it is already defined in the global of this file*/
+      
+      const topic = 'Searching Spotify\'s API.'
+
+      const mainTerm = encodeURI(term); // this converts the term passed as a parameter to a URI format
+
+      let response;
+      let JSONResponse;
+      let error;
+      let mappedTracks;
 
       if (!userAccessToken){
         console.log('No Access Token.');
-        return;
+        return ;
       };
-
-      const mainTerm = encodeURI(term); // this converts the term passed as a parameter to a URI format
-      
       
+      //The code below tries to contact Spotifies API
       try {
-        const response =  await fetch(`https://api.spotify.com/v1/search?type=track&q=${mainTerm}`, {
+        response =  await fetch(`https://api.spotify.com/v1/search?type=track&q=${mainTerm}`, {

```

Comment:
you should store most of the url of spotify in a const

like this

```
const SPOTIFY_SEARCH_ENDPOINT = 'https://api.spotify.com/v1/search?type=track'

// later in code assemble the path
const path = `${SPOTIFY_SEARCH_ENDPOINT}&q=${mainTerm}`

fetch(path)
```

Reply:
Did that and stored it in the a global variable.

Reply:
I stored the repeating URL `https://api.spotify.com/v1` in a `const` in the global scope of the file.
Then I refractored all the code accordingly and it works fine.
Thanks for that advice :)